worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(fishing_tasks)
+        else:
+            fishing_tasks.reverse()
+        fishing_tasks = fishing_tasks[0:self.multiworld.max_fishing_tasks[self.player]]
+        fishing_weight = self.multiworld.fishing_task_weight[self.player] if len(fishing_tasks) > 0 else 0
+
+        cooking_tasks = [task for task in self.locations_by_category["Cooking"]
+                         if task.skills[0].level <= int(self.multiworld.max_cooking_level[self.player])]
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(cooking_tasks)
+        else:
+            cooking_tasks.reverse()
+        cooking_tasks = cooking_tasks[0:self.multiworld.max_cooking_tasks[self.player]]
+        cooking_weight = self.multiworld.cooking_task_weight[self.player] if len(cooking_tasks) > 0 else 0
+
+        firemaking_tasks = [task for task in self.locations_by_category["Firemaking"]
+                            if task.skills[0].level <= int(self.multiworld.max_firemaking_level[self.player])]
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(firemaking_tasks)
+        else:
+            firemaking_tasks.reverse()
+        firemaking_tasks = firemaking_tasks[0:self.multiworld.max_firemaking_tasks[self.player]]
+        firemaking_weight = self.multiworld.firemaking_task_weight[self.player] if len(firemaking_tasks) > 0 else 0
+
+        woodcutting_tasks = [task for task in self.locations_by_category["Woodcutting"]
+                             if task.skills[0].level <= int(self.multiworld.max_woodcutting_level[self.player])]
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(woodcutting_tasks)
+        else:
+            woodcutting_tasks.reverse()
+        woodcutting_tasks = woodcutting_tasks[0:self.multiworld.max_woodcutting_tasks[self.player]]
+        woodcutting_weight = self.multiworld.woodcutting_task_weight[self.player] if len(woodcutting_tasks) > 0 else 0
+
+        all_tasks = [
+            combat_tasks,
+            prayer_tasks,
+            magic_tasks,
+            runecraft_tasks,
+            crafting_tasks,
+            mining_tasks,
+            smithing_tasks,
+            fishing_tasks,
+            cooking_tasks,
+            firemaking_tasks,
+            woodcutting_tasks,
+            general_tasks
+        ]
+        all_weights = [
+            combat_weight,
+            prayer_weight,
+            magic_weight,
+            runecraft_weight,
+            crafting_weight,
+            mining_weight,
+            smithing_weight,
+            fishing_weight,
+            cooking_weight,
+            firemaking_weight,
+            woodcutting_weight,
+            general_weight
+        ]
+
+        while locations_added < locations_required:
+            taskChosen = False
```

Comment:
This variable name also violates PEP8 guidelines btw. I saw a few of those in the code. Are you using an IDE with a built in linter? :)

Reply:
Stupid C# brain